substrate/frame/support/src/traits/tasks.rs

```diff
@@ -0,0 +1,82 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Contains the [`Task`] trait, which defines a general-purpose way for defining and executing
+//! service work, and supporting types.
+
+/// Contain's re-exports of all the supporting types for the [`Task`] trait. Used in the macro
+/// expansion of `RuntimeTask`.
+pub mod prelude {
+	pub use codec::FullCodec;
+	pub use scale_info::TypeInfo;
+	pub use sp_runtime::DispatchError;
+	pub use sp_std::{fmt::Debug, iter::Iterator, vec, vec::IntoIter};
+	pub use sp_weights::Weight;
+}
+
+use prelude::*;
```

Comment:
This is not good practice, since we pollute everything with SemVer abiding re-exports.

Reply:
fair, IIRC this only gets used as the first line of some of the generated functions and never gets used in a user scope, but fair enough